### PR TITLE
allowed single process/zmq_context communication (inproc)

### PR DESCRIPTION
**Pull request by arne123**
added the possibility to use inproc communication for the RPC Server and Client when used in a single process and zmq context

### CODE REVIEW

File: Phoniebox/rpc/PhonieboxRpcClient.py [hunk 1]
```diff
@@ -4,17 +4,23 @@
...
         self.context = None
         
-    def connect(self,addr= None):
-        if addr == None:
-            addr = "tcp://127.0.0.1:5555"
-        self.context = zmq.Context()
+    def connect(self,addr=None,zmq_context=None):
+        if zmq_context != None:
+            self.context = zmq_context
+            local_addr = "inproc://PhonieboxRpcServer"
```
Comment (pabera) [inline Phoniebox/rpc/PhonieboxRpcClient.py:12]: Do we want to outsource this into a config file so we can reference it in different parts of the python app?

File: Phoniebox/rpc/PhonieboxRpcClient.py [hunk 2]
```diff
@@ -4,17 +4,23 @@
...
+        else:
+            self.context = zmq.Context()
+            local_addr = "tcp://127.0.0.1:5555"
```
Comment (pabera) [inline Phoniebox/rpc/PhonieboxRpcClient.py:15]: Same here

File: Phoniebox/rpc/PhonieboxRpcClient.py [hunk 3]
```diff
@@ -25,8 +31,9 @@ def enqueue(self, request):
 
         try:
             server_response = self.queue.recv()
-        except:
-            print ("somethng went wrong")
+        except Exception as e:
+            print ("somethng went wrong:")
```
Comment (pabera) [inline Phoniebox/rpc/PhonieboxRpcClient.py:35]: I might be picky, but if we use something like "[ERROR] ...", it'll be easier for use to search and browser logs if required

File: Phoniebox/rpc/PhonieboxRpcServer.py [hunk 4]
```diff
@@ -13,13 +13,17 @@ def __init__(self,objects):
         self.context = None
         self._keep_running = True
         
-    def connect(self,addr= None):
-        if addr == None:
-            addr = "tcp://127.0.0.1:5555"
+    def connect(self,addrs= None):
+        if addrs == None:
+            addrs = ["tcp://127.0.0.1:5555","inproc://PhonieboxRpcServer"]
```
Comment (pabera) [inline Phoniebox/rpc/PhonieboxRpcServer.py:18]: we could use the variables from above here ;-)

File: docs/phoniebox.conf.example [hunk 5]
```diff
@@ -0,0 +1,17 @@
+[SYSTEM]
+BOX_NAME: PhonieBox
+
+[PLAYER]
+AUDIOFOLDERSPATH:"/home/pi/RPi-Jukebox-RFID/shared/"
+PLAYLISTSFOLDERPATH:"/home/pi/RPi-Jukebox-RFID/playlists"
+MUSIC_PLAYER_STATUS: /home/pi/RPi-Jukebox-RFID/shared/music_player_status.json
```
Comment (pabera) [inline docs/phoniebox.conf.example:7]: To quote, or not to quote, this is the question 😄

File: Phoniebox/rpc/PhonieboxRpcClient.py [hunk 6]
```diff
@@ -4,17 +4,23 @@
...
     
     def __init__(self):
-        #self.objects = objects
         self.context = None
         
-    def connect(self,addr= None):
-        if addr == None:
-            addr = "tcp://127.0.0.1:5555"
-        self.context = zmq.Context()
+    def connect(self,addr=None,zmq_context=None):
```
Comment (ChisSoc) [inline Phoniebox/rpc/PhonieboxRpcClient.py:9]: Any particular reason why you don't connect directly in the constructor?

**Review by ChisSoc — APPROVED** [review]

File: Phoniebox/rpc/PhonieboxRpcClient.py (hunk 3 above)
Comment (arne123) [reply Phoniebox/rpc/PhonieboxRpcClient.py:35]: sure, error handling has not yet been focused on

File: Phoniebox/rpc/PhonieboxRpcClient.py (hunk 6 above)
Comment (arne123) [reply Phoniebox/rpc/PhonieboxRpcClient.py:9]: Yes, the clients are instantiated before the server is running in order to allow a parsing of the clients objects which need to be passed to the server.

File: Phoniebox/rpc/PhonieboxRpcClient.py (hunk 1 above)
Comment (arne123) [reply Phoniebox/rpc/PhonieboxRpcClient.py:12]: It is prepared for passing the address, this just defining the default

File: docs/phoniebox.conf.example (hunk 5 above)
Comment (arne123) [reply docs/phoniebox.conf.example:7]: sure, the quoted one has no yet been used, but lets look forward to yaml...

**Pull request merged**